### DISCUSSION TITLE
what happens or should happen in intercept when unknown types are received ?

### DISCUSSION THREAD

**Discussion by blankdots** [neicnordic/sensitive-data-archive#783] [Q&A] [answered]
we receive in `intercept` services messages of type `dac.dataset` and `dac`, and we get an error
```
time="2024-04-02T10:15:28Z" level=error msg="Exception (406) Reason: \"PRECONDITION_FAILED - delivery acknowledgement on channel 1 timed out. Timeout value used: 1800000 ms. This timeout value can be configured, see consumers doc guide to learn more\""
```

should the message go to an error queue if it is not recognized ?

version is `ghcr.io/neicnordic/sensitive-data-archive:v0.2.59`

**Comment by jbygdell** [comment]
It should probably be sent to `catch_all.dead` and not `error` as it is not routable due to not being implemented.

The other side of this question is: What are we going to do with these messages?
It's not like we have been given any reason for us getting them.

**Reply by blankdots** [reply]
> What are we going to do with these messages?

no good ideas, but maybe just ack them for now, and do nothing, because i think the error will persist.

**Reply by jbygdell** [reply, marked as answer]
ACK and send them to `catch_all.dead`, then we will see if we get any more new messages but they will not disturb the current functionality.

**Reply by kjellp** [reply]
We've also received such a message. Exactly 30 mins after receiving the message of type "dac.dataset", we get the "Exception (406) PRECONDITION_FAILED". And: it causes the intercept container to exit. Other messages is received and routed correctly during these 30 mins by intercept, but then it abruptly stops.

**Reply by blankdots** [reply]
@kjellp some proposal how to handle it for now is in #784 and a follow-up issue in #786

**Comment by blankdots** [comment]
follow-up issue: https://github.com/neicnordic/sensitive-data-archive/issues/786
PR to fix this: https://github.com/neicnordic/sensitive-data-archive/pull/784
